planner/core/resolve_indices.go

```diff
@@ -655,7 +655,19 @@ func (p *PhysicalApply) ResolveIndices() (err error) {
 
 // ResolveIndices implements Plan interface.
 func (p *PhysicalTableScan) ResolveIndices() (err error) {
-	return p.physicalSchemaProducer.ResolveIndices()
+	err = p.physicalSchemaProducer.ResolveIndices()
+	if err != nil {
+		return err
+	}
+	return p.ResolveIndicesItself()
+}
+
+// ResolveIndicesItself implements PhysicalTableScan interface.
+func (p *PhysicalTableScan) ResolveIndicesItself() (err error) {
+	for i, column := range p.schema.Columns {
+		column.Index = i
+	}
+	return
```

Comment:
Is it possible to add UT for it?

Reply:
No, This attribute is mainly used for tiflash, so it cannot be tested by only writing tidb single test.
But after I do a full test, I can see how to add it to the daily regression test.